rclcpp/include/rclcpp/time_utils.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__TIME_UTILS_HPP_
+#define RCLCPP__TIME_UTILS_HPP_
+
+#include <stdexcept>
+#include <type_traits>
+
+#include "rcl/time.h"
+
+namespace rclcpp
+{
+
+/** \defgroup time_conversions Group contains conversion helper functions
+ *  @{
+ */
+
+/// Combines signed seconds and unsigned nanoseconds into one field
+/**
+ * Given seconds and nanoseconds the function merges both into one field.
+ *
+ * Pseudo-code:
+ * output = sec * 1e10 + nanoseconds % 1e10
+ *
+ * \param sec seconds
+ * \param nano nanoseconds
+ * \return field containing seconds and nanoseconds
+ */
+inline int64_t join_to_nano(int32_t sec, uint32_t nano) noexcept
```

Comment:
I feel like this function should be a new macro in here: https://github.com/ros2/rcutils/blob/master/include/rcutils/time.h

Reply:
i would disagree with that, since the operation is strongly type dependent and there is no way for rcutils to know the type of its client libraries. This is also an argument against putting this logic into a marco itself

Reply:
it doesn't necessarily have to be a macro and could also be a free function to save the types. I am just saying that I could see other libraries below `rclcpp` making use of this logic.